src/main/scala/com/gu/tip/assertion/TipAssert.scala

```diff
@@ -0,0 +1,53 @@
+package com.gu.tip.assertion
+
+import java.util.concurrent.Executors
+
+import com.typesafe.scalalogging.{LazyLogging, Logger}
+import retry.Defaults
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Random, Success, Try}
+
+sealed trait AssertionResult
+case object TipAssertPass extends AssertionResult
+case object TipAssertFail extends AssertionResult
+
+trait TipAssertIf {
+  protected val logger: Logger
+
+  val threadPool = Executors.newFixedThreadPool(
+    2,
```

Comment:
```suggestion
    numThreads = 2,
```
I'm not sure what the 2 means, so consider a named parameter? ( I illustrated above but the name is a guess!)